service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsController.java

```diff
@@ -0,0 +1,54 @@
+package uk.gov.hmcts.reform.fpl.controllers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.swagger.annotations.Api;
+import org.assertj.core.util.Lists;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.common.C2DocumentBundle;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+@Api
+@RestController
+@RequestMapping("/callback/upload-c2")
+public class UploadC2DocumentsController {
+    private final ObjectMapper mapper;
+
+    @Autowired
+    private UploadC2DocumentsController(ObjectMapper mapper) {
+        this.mapper = mapper;
+    }
+
+    @PostMapping("/about-to-submit")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody CallbackRequest callbackrequest) {
+        Map<String, Object> data = callbackrequest.getCaseDetails().getData();
+        CaseData caseData = mapper.convertValue(data, CaseData.class);
+
+        data.put("C2DocumentBundle", buildC2DocumentBundle(caseData));
+        data.remove("temporaryC2Document");
+
+        return AboutToStartOrSubmitCallbackResponse.builder().data(data).build();
+    }
+
+    private List<Element<C2DocumentBundle>> buildC2DocumentBundle(CaseData caseData) {
+        List<Element<C2DocumentBundle>> c2DocumentBundle = caseData.getC2DocumentBundle() == null
+            ? Lists.newArrayList() : caseData.getC2DocumentBundle();
```

Comment:
I find some of theser in line if statements hard to read. defaultIfNull(caseData,.getC2DocBundle, newArrayList()) should work here instead